### PR TITLE
Add support for attribute passes

### PR DESCRIPTION
**Pull request by leonardt**
Enables support for `m.combinational2` as a pass.  It augments the get_name logic to support basic attributes by recursing through the tree.

### CODE REVIEW

File: ast_tools/passes/util.py [hunk 1]
```diff
@@ -82,14 +81,15 @@ def set_decorators(tree, decorators):
...
-    @staticmethod
-    def get_name(node):
+    @classmethod
+    def lookup(cls, node, env):
         if isinstance(node, ast.Call):
-            assert isinstance(node.func, ast.Name)
-            return node.func.id
+            return cls.lookup(node.func, env)
+        elif isinstance(node, ast.Attribute):
+            return getattr(cls.lookup(node.value, env), node.attr)
```
Comment (cdonovick) [inline ast_tools/passes/util.py:89]: Does this need to be done recursively for `node.attr1.attr2`?

File: ast_tools/passes/util.py [hunk 2]
```diff
@@ -103,14 +103,17 @@ def set_decorators(tree, decorators):
...
-            assert isinstance(node.decorator.func, cst.Name)
-            return node.decorator.func.value
+    @classmethod
+    def lookup(cls, node: cst.CSTNode, env):
+        if isinstance(node, cst.Decorator):
+            return cls.lookup(node.decorator, env)
+        elif isinstance(node, cst.Call):
+            return cls.lookup(node.func, env)
+        elif isinstance(node, cst.Attribute):
+            return getattr(cls.lookup(node.value, env), node.attr.value)
```
Comment (cdonovick) [inline ast_tools/passes/util.py:113]: same as above

**Comment by cdonovick** [comment]
Also could you add a test

**Comment by leonardt** [comment]
There is a test: `test_apply_attribute`, unless you wanted it to test something else specifically?

File: ast_tools/passes/util.py (hunk 1 above)
Comment (leonardt) [reply ast_tools/passes/util.py:89]: node.value will be itself an attribute node, so the recursive call to `lookup` will handle it, unless you're asking if we should not handle the recursive attribute case?

**Comment by leonardt** [comment]
Oh I see, my test doesn't actually test the right case (this is only used in the begin/end rewrite pattern), will fix it

**Comment by leonardt** [comment]
Updated the test (just reused the debug test) and confirmed it works in the recursive case.

**Pull request merged**